### PR TITLE
Fix issue with feedback form replacement and add acceptance tests

### PR DESCRIPTION
**Pull request by virginiacc**
A couple of Notification module references in `FormSubmit.js` are breaking feedback form replacement on production -- the form can be submitted, but an error prevents its replacement with a success notification, leaving a large blank space where the form had been (see an [Ask CFPB question](https://www.consumerfinance.gov/ask-cfpb/i-would-like-to-be-able-to-have-my-friend-or-family-member-help-with-my-bill-paying-and-banking-what-are-my-options-en-1145/) for an example). This PR updates the references and adds acceptance tests for feedback form notifications and replacement.

## Additions

- Acceptance tests for feedback form replacement

## Changes

- Notification module references in `FormSubmit.js`

## Testing

1. Check out branch and run `gulp scripts`
2. Navigate to [an Ask CFPB answer page](http://localhost:8000/ask-cfpb/i-would-like-to-be-able-to-have-my-friend-or-family-member-help-with-my-bill-paying-and-banking-what-are-my-options-en-1145/) locally and verify that its feedback form is replaced with a success message when submitted with a radio option selected
3. Run `gulp test:acceptance` and verify that tests pass

### Production
screen shot 2019-01-15 at 9 23 18 pm

### Local

screen shot 2019-01-15 at 9 26 17 pm

## Todos

- Form is still displayed after page refresh when feedback is successfully submitted with javascript disabled

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Internet Explorer 8, 9, 10, and 11
- [ ] Edge
- [ ] iOS Safari
- [ ] Chrome for Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: cfgov/unprocessed/js/organisms/FormSubmit.js [hunk 1]
```diff
@@ -29,7 +28,7 @@ function FormSubmit( element, baseClass, opts ) {
   const _baseElement = checkDom( element, baseClass );
   const _formElement = _baseElement.querySelector( 'form' );
   const _notificationElement = _baseElement.querySelector(
-    Notification.BASE_CLASS
+    `.${ Notification.BASE_CLASS }`
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/FormSubmit.js]: Whoops

File: test/browser_tests/cucumber/step_definitions/feedback-form.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+const SUBMIT_BUTTON = BASE_SEL + ' [type="submit"]';
+const NOTIFICATION = BASE_SEL + ' .m-notification';
+const SUCCESS_CLASS = 'm-notification__success';
+const ERROR_CLASS = 'm-notification__error';
+
+chai.use( chaiAsPromised );
+
+let _dom;
+
+Before( function() {
```
Comment (anselmbradford) [inline test/browser_tests/cucumber/step_definitions/feedback-form.js]: TRIVIAL: You could use arrow functions here and throughout to make the tests more compact `() => { …`

File: test/browser_tests/cucumber/step_definitions/feedback-form.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
+const { Then, When, Before } = require( 'cucumber' );
+const chai = require( 'chai' );
+const expect = chai.expect;
```
Comment (anselmbradford) [inline test/browser_tests/cucumber/step_definitions/feedback-form.js]: TRIVIAL: Probably could get rid of line 2 by using an import here `import { expect } from 'chai'`

File: cfgov/unprocessed/js/organisms/FormSubmit.js [hunk 4]
```diff
@@ -162,7 +161,7 @@ function FormSubmit( element, baseClass, opts ) {
     }
 
     function fadeInMessage() {
-      _notification.update( _notification.SUCCESS, message );
+      _notification.update( Notification.SUCCESS, message );
```
Comment (anselmbradford) [inline cfgov/unprocessed/js/organisms/FormSubmit.js]: Same issue in https://github.com/cfpb/cfgov-refresh/pull/4770. I swear I updated this but I think I rebased it wrong before merging :(

**Comment by virginiacc** [comment]
@anselmbradford Thanks for the feedback on the tests! I'm going to merge this now to deploy the fix & address that in a separate PR.

**Pull request merged**